mls-rs/src/group/external_commit.rs

```diff
@@ -168,20 +137,9 @@ impl<C: ClientConfig> ExternalCommitBuilder<C> {
     /// The authenticity of the proposal is NOT fully verified. It is only verified the
     /// same way as by [`ExternalGroup`](`crate::external_client::ExternalGroup`).
     /// The proposal MUST be an MlsPlaintext, else the [`Self::build`] function will fail.
-    pub fn with_received_custom_proposal(mut self, proposal: MlsMessage) -> Result<Self, MlsError> {
-        let MlsMessagePayload::Plain(plaintext) = proposal.payload else {
-            return Err(MlsError::UnexpectedMessageType);
-        };
-
-        let super::Content::Proposal(proposal) = plaintext.content.content.clone() else {
-            return Err(MlsError::UnexpectedMessageType);
-        };
-
-        // We store proposal to verify authenticity later. At this point this may not be possible if we
-        // don't have the tree.
-        self.received_custom_proposals.push(plaintext);
-
-        Ok(self.with_proposal(*proposal))
+    pub fn with_received_custom_proposal(mut self, proposal: MlsMessage) -> Self {
+        self.received_custom_proposals.push(proposal);
+        self
     }
```

Comment:
How does this work with custom proposal rules being applied outside? I forgot to add a function for this but it's like in CommitBuilder:
```handle_custom_proposals(commit_builder.proposals_mut());  // expand custom proposals, add more proposals if needed```